tools/toolchain/optimized_clang.sh

```diff
@@ -146,7 +146,7 @@ if [[ "${CLANG_BUILD}" = "INSTALL" ]]; then
     ninja -C build
 
     cd "${CLANG_ROOT_DIR}"
-    rm -rf "${CLANG_BUILD_DIR}"/{build/profiles,*.prof,prof.fdata}
+    rm -rf "${CLANG_BUILD_DIR}"/{build/profiles,ir.prof,csir.prof,prof.fdata}
```

Comment:
Surprised, I was able to install the optimized clang on my machine.

Why is combined.prof necessary?

Reply:
> Surprised, I was able to install the optimized clang on my machine.

I think I was also able to install optimized clang when I tested clang-18 patch, but now I cannot able to reproduce it.
 
> Why is combined.prof necessary?

combined.prof is required because ninja is trying to re-compile clang for some reason, so the real problem is unnecessary re-compile, not missing combined.prof.
So I now I think this patch is wrong.

I tried to use debug option of ninja to find out why it's recompiling.
And it seems like because the header of standard library updated by `yum update`:
```
+ ninja -C build -v -j1 -dexplain install-distribution-stripped
ninja: Entering directory `build'
ninja explain: output tools/clang/lib/Support/CMakeFiles/obj.clangSupport.dir/RISCVVIntrinsicUtils.cpp.o older than most recent input /usr/include/linux/errno.h (1720616913000000000 vs 1722816000000000000)
ninja explain: tools/clang/lib/Support/CMakeFiles/obj.clangSupport.dir/RISCVVIntrinsicUtils.cpp.o is dirty
ninja explain: output tools/clang/utils/TableGen/CMakeFiles/clang-tblgen.dir/ASTTableGen.cpp.o older than most recent input /usr/include/linux/errno.h (1720616911000000000 vs 1722816000000000000)
ninja explain: tools/clang/utils/TableGen/CMakeFiles/clang-tblgen.dir/ASTTableGen.cpp.o is dirty
ninja explain: output tools/clang/utils/TableGen/CMakeFiles/clang-tblgen.dir/ClangASTNodesEmitter.cpp.o older than most recent input /usr/include/linux/errno.h (1720616912000000000 vs 1722816000000000000)
ninja explain: tools/clang/utils/TableGen/CMakeFiles/clang-tblgen.dir/ClangASTNodesEmitter.cpp.o is dirty
ninja explain: output tools/clang/utils/TableGen/CMakeFiles/clang-tblgen.dir/ClangASTPropertiesEmitter.cpp.o older than most recent input /usr/include/linux/errno.h (1720616912000000000 vs 1722816000000000000)
ninja explain: tools/clang/utils/TableGen/CMakeFiles/clang-tblgen.dir/ClangASTPropertiesEmitter.cpp.o is dirty
ninja explain: output tools/clang/utils/TableGen/CMakeFiles/clang-tblgen.dir/ClangAttrEmitter.cpp.o older than most recent input /usr/include/linux/errno.h (1720616916000000000 vs 1722816000000000000)
ninja explain: tools/clang/utils/TableGen/CMakeFiles/clang-tblgen.dir/ClangAttrEmitter.cpp.o is dirty
ninja explain: output tools/clang/utils/TableGen/CMakeFiles/clang-tblgen.dir/ClangCommentCommandInfoEmitter.cpp.o older than most recent input /usr/include/linux/errno.h (1720616912000000000 vs 1722816000000000000)
ninja explain: tools/clang/utils/TableGen/CMakeFiles/clang-tblgen.dir/ClangCommentCommandInfoEmitter.cpp.o is dirty
ninja explain: output tools/clang/utils/TableGen/CMakeFiles/clang-tblgen.dir/ClangCommentHTMLNamedCharacterReferenceEmitter.cpp.o older than most recent input /usr/include/linux/errno.h (1720616912000000000 vs 1722816000000000000)
ninja explain: tools/clang/utils/TableGen/CMakeFiles/clang-tblgen.dir/ClangCommentHTMLNamedCharacterReferenceEmitter.cpp.o is dirty
ninja explain: output tools/clang/utils/TableGen/CMakeFiles/clang-tblgen.dir/ClangCommentHTMLTagsEmitter.cpp.o older than most recent input /usr/include/linux/errno.h (1720616912000000000 vs 1722816000000000000)
ninja explain: tools/clang/utils/TableGen/CMakeFiles/clang-tblgen.dir/ClangCommentHTMLTagsEmitter.cpp.o is dirty
ninja explain: output tools/clang/utils/TableGen/CMakeFiles/clang-tblgen.dir/ClangDataCollectorsEmitter.cpp.o older than most recent input /usr/include/linux/errno.h (1720616912000000000 vs 1722816000000000000)
```

I guess it's bit difficult to avoid such recompile, and we don't really need to keep  whole clang build directory for a optimized clang archive.
We should archive clang install image instead.
To do so, we can use                                                                                                                                   
 `DESTDIR=<destdir> ninja install-distribution-stripped`, and archive <destdir> as clang archive.

Reply:
Pushed new version of the patch which implemented this.